packages/client/modules/userDashboard/components/OrgBilling/OrgPlanDrawer.tsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react'
+import styled from '@emotion/styled'
+import graphql from 'babel-plugin-relay/macro'
+import {commitLocalUpdate, useFragment} from 'react-relay'
+import {OrgPlanDrawer_organization$key} from '../../../../__generated__/OrgPlanDrawer_organization.graphql'
+import ResponsiveDashSidebar from '../../../../components/ResponsiveDashSidebar'
+import {Close} from '@mui/icons-material'
+import {PALETTE} from '../../../../styles/paletteV3'
+import {BezierCurve, Breakpoint, DiscussionThreadEnum, ZIndex} from '../../../../types/constEnums'
+import LabelHeading from '../../../../components/LabelHeading/LabelHeading'
+import {desktopSidebarShadow} from '../../../../styles/elevation'
+import useBreakpoint from '../../../../hooks/useBreakpoint'
+import PlainButton from '../../../../components/PlainButton/PlainButton'
+import useAtmosphere from '../../../../hooks/useAtmosphere'
+import {ICON_SIZE} from '../../../../styles/typographyV2'
+import OrgPlanDrawerContent from './OrgPlanDrawerContent'
+
+const DrawerHeader = styled('div')({
+  alignItems: 'center',
+  display: 'flex',
+  justifyContent: 'space-between',
+  padding: '16px 8px 16px 16px',
+  width: '100%'
+})
+
+const Drawer = styled('div')<{isDesktop: boolean; isOpen: boolean}>(({isDesktop, isOpen}) => ({
+  boxShadow: isDesktop ? desktopSidebarShadow : undefined,
+  backgroundColor: '#FFFFFF',
+  display: 'flex',
+  flex: 1,
+  flexDirection: 'column',
+  justifyContent: 'stretch',
+  overflow: 'hidden',
+  position: isDesktop ? 'fixed' : 'static',
+  bottom: 0,
+  top: 0,
+  right: isDesktop ? 0 : undefined,
+  transition: `all 200ms ${BezierCurve.DECELERATE}`,
+  userSelect: 'none',
+  transform: `translateX(${isOpen ? 0 : DiscussionThreadEnum.WIDTH}px)`,
+  width: DiscussionThreadEnum.WIDTH,
+  zIndex: ZIndex.SIDEBAR,
+  height: '100%',
+  '@supports (height: 1svh) and (height: 1lvh)': {
+    height: isDesktop ? '100lvh' : '100svh'
+  }
+}))
+
+const CloseIcon = styled(Close)({
+  color: PALETTE.SLATE_600,
+  cursor: 'pointer',
+  '&:hover': {
+    opacity: 0.5
+  }
+})
+
+const StyledCloseButton = styled(PlainButton)({
+  height: ICON_SIZE.MD18,
+  display: 'flex',
+  alignItems: 'center'
+})
+
+const StyledLabelHeading = styled(LabelHeading)({
+  fontSize: 12,
+  lineHeight: '18px',
+  textTransform: 'none'
+})
+
+type Props = {
+  organizationRef: OrgPlanDrawer_organization$key
+}
+
+const OrgPlanDrawer = (props: Props) => {
+  const {organizationRef} = props
+  const organization = useFragment(
+    graphql`
+      fragment OrgPlanDrawer_organization on Organization {
+        id
+        showDrawer
+        tier
+      }
+    `,
+    organizationRef
+  )
+  const {id: orgId, tier, showDrawer} = organization
+  const atmosphere = useAtmosphere()
+  const isDesktop = useBreakpoint(Breakpoint.SIDEBAR_LEFT)
```

Comment:
+1 I like reusing breakpoints, but here it seems this is not the best one. Because we disable vertical scrolling, it would seem more appropriate to me if we'd gray out everything as soon as the details would not fit anymore.
Gray at sidebar left breakpoint:
![Screenshot from 2023-04-11 11-10-45](https://user-images.githubusercontent.com/7331043/231113513-74e4afa8-d708-424d-82c9-da5662d7869e.png)

Not gray when details don't hide the `Plans`:
![Screenshot from 2023-04-11 11-11-14](https://user-images.githubusercontent.com/7331043/231113738-5d143b8c-304f-430b-849e-937bfb1512b4.png)

The edge case between where some part of the Plans is hidden, we cannot scroll vertically and it's not grayed out:
![Screenshot from 2023-04-11 11-11-01](https://user-images.githubusercontent.com/7331043/231113887-48a46646-e3b6-44e0-9ab5-2159175b5857.png)

Reply:
@acressall here's a Loom on what I currently see: https://www.loom.com/share/194140ccc25344feb4e67b13e30d2cf5

@Dschoordsch mentioned that we cannot scroll vertically when some part of the Plans is hidden, but I haven't been able to reproduce that. However, the draw can cover some of the content. Alicia, do you think we should grey out the background when the draw covers any content or keep things as they are now? You can see how it currently looks in the Loom

Reply:
Thanks so much for the Loom! So the difference with the agenda panel is it's over a task board so we have a horizontal scroll behind that lets you see the rest of the content. I'm leaning towards adding the grey background as soon as it covers the content, for the sake of contrast, so anyone with a visual disability (or even just bad screen calibration) can distinguish the panel from the background content

Reply:
Makes sense! I've updated the breakpoints for the drawer. Here's a Loom with the update: https://www.loom.com/share/00b2462577a14e178c4835de97388eac

What do you think @acressall?

Reply:
Looks great! Thanks Nick!